ros2cli/ros2cli/node/daemon.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import platform
+import socket
+import subprocess
+
+from ros2cli.daemon import PORT
+
+import xmlrpc.client
```

Comment:
Using `XML-RPC` gets us going quickly since it is very easy to use in Python.

In the future it might be replaced with any other protocol. E.g. even with DDS once we are able to connect to a service without the overhead of the discovery.